src/test/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacadeIT.java

```diff
@@ -432,4 +437,35 @@ public void assignBulkEndpoint_scheduleSingleAssignmentWithScheduledDateAfterDue
         // check status code is NO_CONTENT (successful)
         assertEquals(Response.Status.NO_CONTENT.getStatusCode(), deleteAssignmentResponse.getStatus());
     }
+
+    @Test public void getAssignmentProgressDownloadCSV_succeeds() throws Exception {
+        // log in as Test teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, ITConstants.TEST_TEACHER_EMAIL,
+                ITConstants.TEST_TEACHER_PASSWORD);
+        HttpServletRequest downloadAssignmentRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(downloadAssignmentRequest);
+
+        Response downloadAssignmentResponse = assignmentFacade.getAssignmentProgressDownloadCSV(downloadAssignmentRequest, 4L, "excel");
+        String downloadAssignmentContents = downloadAssignmentResponse.getEntity().toString();
+        
+        // Note that the first two lines have a trailing whitespace. If your IDE automatically removes these from the expected file it may cause a mismatch
+        try (FileInputStream expectedFile = new FileInputStream("src/test/resources/expected_assignment_export.csv");) {
```

Comment:
No need for the semicolon.

Reply:
Actually, is the `try` necessary at all?

Reply:
The CodeQL analysis highlighted this as a potential issue on a previous commit and recommended using try-with-resources to ensure it was closed after use.
https://github.com/isaaccomputerscience/isaac-api/security/code-scanning/338

Reply:
I see, thanks for clarifying. Please not that just adding a `try` is not enough, the problem is still there (you're not closing the resource, you need to call `expectedFile.close()`, ideally in a `finally` block). You could move all this to its own function that returns the contents as a string. Or alternatively, you could try a different way of reading a file, something like:
```java
String expectedContents = new String(Files.readAllBytes(Paths.get("src/test/resources/expected_assignment_progress_export.csv")));
```

Reply:
Just for future reference, as pointed out by Robert, I was wrong about the need to close it: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html